src/namedtuple/productnamedtuple.jl

```diff
@@ -0,0 +1,131 @@
+"""
+    ProductNamedTupleDistribution{Tnames,Tdists,S<:ValueSupport,eltypes} <:
+        Distribution{NamedTupleVariate{Tnames},S}
+
+    A distribution of `NamedTuple`s, constructed from a `NamedTuple` of independent named
+    distributions.
+
+    Users should use [`product_distribution`](@ref) to construct a product distribution of
+    independent distributions instead of constructing a `ProductNamedTupleDistribution`
+    directly.
+"""
+struct ProductNamedTupleDistribution{Tnames,Tdists,S<:ValueSupport,eltypes} <:
+       Distribution{NamedTupleVariate{Tnames},S}
+    dists::NamedTuple{Tnames,Tdists}
+end
+function ProductNamedTupleDistribution(
+    dists::NamedTuple{K,V}
+) where {K,V<:Tuple{Vararg{Distribution}}}
```

Comment:
```suggestion
) where {K,V<:Tuple{Distribution,Vararg{Distribution}}}
```
to ensure that there is at least one distribution